common/thrift_router.h

```diff
@@ -377,9 +382,20 @@ class ThriftRouter {
         const Role role,
         const unsigned rotation_counter,
         const std::string& segment,
-        const int shrink_target) {
+        const int shrink_target,
+        const std::string& specific_az) {
       std::vector<const Host*> v;
-      if (role == Role::ANY && !FLAGS_always_prefer_local_host) {
+      if (specific_az != "") {
+        // if az is specified by caller, return hosts belong only to that az.
+        for (const auto& hi : host_info) {
+          if (hi.first->az.compare(specific_az) == 0) {
+            v.push_back(hi.first);
```

Comment:
what if there is no matched hosts for the AZ?

Reply:
if there are no matched hosts, no clients will be returned as user explicitly wants the clients of the az specified